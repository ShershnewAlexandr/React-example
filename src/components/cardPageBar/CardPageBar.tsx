import React from 'react';
import { Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import { CardPageBarContainerProps } from '../../containers/CardPageBarContainer';
import './CardPageBarStyles.css';

function CardPageBar(props: CardPageBarContainerProps) {
  const { nextPageAction, backPageAction, currentPage, lastPage } = props;
  const backButtonDisable = currentPage === 1;
  const nextButtonDisable = currentPage === lastPage;
  const { t } = useTranslation();

  return (
    <Grid container spacing={4}>
      <Grid item sm={4}>
        <button
          className={
            backButtonDisable ? 'disabledButton' : 'card-page-bar-button'
          }
          onClick={backPageAction}
          disabled={backButtonDisable}
        >
          <i className="material-icons arrows-page-bar">arrow_back</i>
        </button>
      </Grid>
      <Grid item sm={4}>
        <Grid container justify="center" alignItems="center">
          <Grid item>
            <span>
              <span>{currentPage}</span>
              {` ${t('cardPageBar.of')}  ${lastPage}`}
            </span>
          </Grid>
        </Grid>
      </Grid>
      <Grid item sm={4}>
        <button
          className={
            nextButtonDisable ? 'disabledButton' : 'card-page-bar-button'
          }
          onClick={nextPageAction}
          disabled={nextButtonDisable}
        >
          <i className="material-icons arrows-page-bar">arrow_forward</i>
        </button>
      </Grid>
    </Grid>
  );
}

export default CardPageBar;
