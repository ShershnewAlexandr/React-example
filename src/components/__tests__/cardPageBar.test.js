import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import CardPageBar from '../cardPageBar/CardPageBar';

jest.mock('../../store/index');
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: str => str,
  }),
}));

describe('cardPageBar', () => {
  const props = {
    current_page: 1,
    total: 4,
    backPageAction: jest.fn(),
    nextPageAction: jest.fn(),
  };
  const cardPageBar = shallow(<CardPageBar {...props} />);

  it('snapshot', () => {
    expect(toJson(cardPageBar)).toMatchSnapshot();
  });
});
