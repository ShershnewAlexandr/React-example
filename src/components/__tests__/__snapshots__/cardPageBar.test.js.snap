// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cardPageBar snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={4}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
  >
    <button
      className="card-page-bar-button"
      disabled={false}
      onClick={[MockFunction]}
    >
      <i
        className="material-icons arrows-page-bar"
      >
        arrow_back
      </i>
    </button>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      justify="center"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <span>
          <span />
           cardPageBar.of  undefined
        </span>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={4}
  >
    <button
      className="disabledButton"
      disabled={true}
      onClick={[MockFunction]}
    >
      <i
        className="material-icons arrows-page-bar"
      >
        arrow_forward
      </i>
    </button>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
